gnome3.20: init at 3.20.0

DamienCassou opened the pull request:
Having fun trying to package Gnome 3.20.

Issue: #14289

cc @lethalman 

---

To be done:
- [x] Aisleriot doesn't build ([upstream bug](https://bugzilla.gnome.org/show_bug.cgi?id=764305), but this is not blocking for us) : fixed by making a libcanberra version depending on `gtk+-3.20`
- [ ] Fix commits with `(todo)` in their message
- [ ] Enable Google authentication in evolution-data-server
- [ ] Make sure  the `nixos-rebuild` output depends on latest libraries:
  - no gtk 3.18
  - no gnome 3.18
  - latest clutter

DamienCassou committed:
Clone 3.18 directory to 3.20

DamienCassou committed:
gtk3: 3.18.5 -> 3.20.0

DamienCassou committed:
Add Gnome 3.20

DamienCassou committed:
Automatically update Gnome packages

This was done by running:

$ ./maintainers/scripts/gnome.sh pkgs/desktops/gnome-3/3.20 update-all 3.20

DamienCassou committed:
Automatically add new Gnome packages

This was done by running:

$ ./maintainers/scripts/gnome.sh pkgs/desktops/gnome-3/3.20 update-all 3.20

DamienCassou committed:
gnome-pomodoro: old-git-rev -> 0.11.2

jgillich commented:
:+1:  I'll try this soon. Any issues you've been running into so far?

DamienCassou commented:
```
/tmp/nix-build-aisleriot-3.20.1.drv-0/aisleriot-3.20.1/src/board-noclutter.c:3423: undefined reference to `gtk_widget_class_set_css_name'
collect2: error: ld returned 1 exit status
Makefile:606: recipe for target 'sol' failed
make[4]: *** [sol] Error 1
make[4]: Leaving directory '/tmp/nix-build-aisleriot-3.20.1.drv-0/aisleriot-3.20.1/src'
Makefile:896: recipe for target 'all-recursive' failed
make[3]: *** [all-recursive] Error 1
make[3]: Leaving directory '/tmp/nix-build-aisleriot-3.20.1.drv-0/aisleriot-3.20.1/src'
Makefile:519: recipe for target 'all' failed
make[2]: *** [all] Error 2
make[2]: Leaving directory '/tmp/nix-build-aisleriot-3.20.1.drv-0/aisleriot-3.20.1/src'
Makefile:497: recipe for target 'all-recursive' failed
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory '/tmp/nix-build-aisleriot-3.20.1.drv-0/aisleriot-3.20.1'
Makefile:428: recipe for target 'all' failed
make: *** [all] Error 2
builder for ‘/nix/store/jc6x2696wd3li1al1mardsz6826ih182-aisleriot-3.20.1.drv’ failed with exit code 2
error: build of ‘/nix/store/jc6x2696wd3li1al1mardsz6826ih182-aisleriot-3.20.1.drv’ failed
```

lethalman commented on pkgs/desktops/gnome-3/3.20/default.nix:
```diff
@@ -0,0 +1,395 @@
+
+  callPackage = pkgs.newScope self;
+
+  version = "3.18";
```
This is wrong.

lethalman commented:
@DamienCassou it's probable that aisleriot is pulling gtk 3.18 instead of gtk 3.20 due to some dependency relying on gtk 3.18. You can find that with `nix-store -qR` of the aisleriot .drv.

DamienCassou committed:
Update Gnome 3 version

lethalman commented:
Also, like for aisleriot, we must be sure that the nixos-rebuild output does not depend on gtk 3.18 or any of the gnome 3.18 dependencies.

DamienCassou committed:
libgames-support: init at 1.0

DamienCassou committed:
gnome-klotski: fix dependencies

DamienCassou committed:
webkitgtk_gtk3_20: init at 2.12

DamienCassou commented:
@lethalman what do you think about adding that to `all-packages.nix`:

```diff
 
+  webkitgtk_gtk3_20 = callPackage ../development/libraries/webkitgtk {
+    harfbuzz = harfbuzz-icu;
+    gst-plugins-base = gst_all_1.gst-plugins-base;
+    gtk3 = gtk3_20;
+  };
+
```

lethalman commented:
@DamienCassou looks ok, what about using `.override` instead?

vcunat commented:
Do you expect some reason to keep both gtk+ 3.18 and 3.20? It would seem much easier to just have one `gtk3` and try to fix (potential) problems with apps using the newer version.

lethalman commented:
@vcunat no there's no reason to keep both versions. The last time it was a huge regression in styling, but I hope it's no more the case. That is, gnome 3.18 should work fine with gtk 3.20, but it must be tested... I wouldn't bet on it.

vcunat commented:
Hmm, the relevant section of [readme](https://git.gnome.org/browse/gtk+/plain/README.in?id=3.20.0) suggests there might be some compatibility issues, e.g. require some CSS theming changes (again). I can't see how serious it might be, but I think we can chance this on master/staging.

DamienCassou commented:
@lethalman can you tell me more about that?
@vcunat I thought it would be a simple first step to do it like this (this is the first time I do something as large as this task). I can do it your way if you think it's easier.

lethalman commented:
This is quite scary "The way theming works in GTK+ has been reworked fundamentally". Btw we did this transition in the past, what about doing it again @vcunat ? We'll do out best to drop gnome 3.18.

vcunat commented:
Hmm, I see there's no mention of theming rework in previous release notes, but I hope this time won't be worse than the previous time.

DamienCassou committed:
glib: 2.46.1 -> 2.48.0

DamienCassou committed:
Gnome now references gtk+3.20 based webkitgtk

DamienCassou committed:
gnome-calculator: add missing dependency

DamienCassou committed:
gspell: init at 1.0.0

DamienCassou committed:
libgit2-glib: 0.23.6 -> 0.24.0

DamienCassou committed:
gedit: add missing dependency

DamienCassou committed:
gnome-robots: add missing dependency

DamienCassou committed:
gpaste: 3.18.3 -> 3.20

DamienCassou committed:
gnome-nibbles: add missing dependencies

DamienCassou committed:
gdk-pixbuf: 2.32.3 -> 2.34.0

DamienCassou committed:
gnome-online-accounts: fix by removing autoreconf

I don't know the original goal of this autoreconf stuff, but it now
breaks the build.

DamienCassou committed:
grilo: 0.2.12 -> 0.3.0

DamienCassou committed:
gnome-control-center: add missing dependency

DamienCassou committed:
cantarell-fonts: 0.0.17 -> 0.0.24

DamienCassou committed:
clutter_1_26: init at 1.26.0

DamienCassou committed:
clutter-gtk-1.8: init at 1.8.0

DamienCassou committed:
clutter-gst: 3.0.4 -> 3.0.18

DamienCassou committed:
gnome3_20: use latest clutter

DamienCassou committed:
json-glib: 1.0.2 -> 1.2.0

lethalman commented:
@DamienCassou did you happen to test gnome 3.18 with gtk 3.20 ?

DamienCassou commented:
@lethalman no, I'm focusing on packagnig gnome 3.20 right now.

lethalman commented:
@DamienCassou ok so I'll try to test gtk 3.20 on gnome 3.18 in the next days.

DamienCassou committed:
(todo) evolution-data-server: disable google auth

I don't want to disable it, it's just that enabling it doesn't work and
I don't know why.

DamienCassou committed:
gnome-weather: add missing dependency

DamienCassou committed:
gobject-introspection: 1.46.0 -> 1.48.0

DamienCassou committed:
(todo) gnome-maps: remove libsoup patch

This patch breaks the build and I have the impression it is not
necessary anymore. I should check.

DamienCassou committed:
(todo) sushi: 3.18.0 -> 3.19.90

Check to see why it doesn't build

DamienCassou commented:
@lethalman is it normal that the number of commits keeps on increasing? I don't see the end :-) (and webkit is getting built for the 5th time).

@k0ral this pull request contains `webkitgtk_gtk3_20` with webkit 2.12.

vcunat commented:
Do you feel a Hydra jobset would help?

DamienCassou commented:
@vcunat I've never worked in coordination with Hydra, I don't know how it would help. Maybe it's time to try :-).

vcunat commented:
I think it will be beset if you push this WIP as gnome-3.20 branch in the *official* nixpkgs repository and I will create a jobset for that, OK? (I trust you do have push access.)

DamienCassou commented:
@vcunat done. What does it change for me? What do I have to do?

vcunat commented:
I created http://hydra.nixos.org/jobset/nixos/gnome-3.20. It gets lower priority than the usual branches so it will be slower, but we should see build breakages in there and get binaries into the official cache immediately.

lethalman commented:
gtk 3.20.2 works well with gnome 3.18. We can drop gtk 3.18 altogether instead of keeping a separate version.

DamienCassou commented:
@vcunat thank you. I will have a look.

@lethalman I'm preparing a rebase of this branch based on your tests. Thanks. Should I do the same for webkitgtk and just update its version?

lethalman commented:
@DamienCassou yes, webkitgtk will just depend upon the new gtk.

DamienCassou commented:
@vcunat sorry, I have no idea how to read http://hydra.nixos.org/jobset/nixos/gnome-3.20. Is there some documentation around?

vcunat commented:
Reading comparisons to master will probably be most useful. http://hydra.nixos.org/eval/1251962?compare=trunk-combined There have been some fixes in there since, leading to showing incorrect "new failures", but probably not too many ATM.

DamienCassou committed:
gnome3.20: add myself to the maintainers

DamienCassou commented:
Move PR to #14366 because of changing remote (from https://github.com/DamienCassou/nixpkgs to https://github.com/NixOS/nixpkgs).
